### PR TITLE
added create room function

### PR DESCRIPTION
**Pull request by ian-tabs**
please review my pull request

### CODE REVIEW

File: frontend/src/states/playerStore.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+const playerStore = create<PlayerStore>((set: SetState<PlayerStore>) => ({
+  id: -1,
+  roomId: '',
+  status: 'standby',
+  points: 0,
+  whiteCards: [],
+  redCards: [],
+  playedRedCard: null,
+  playedWhiteCards: [],
+  setId: (passedPlayerId) => {
```
Comment (jacintojosh) [inline frontend/src/states/playerStore.tsx:13]: Can shorten this to `setId: (id) => set({ id })`
It's okay not to add `passed` to the param name since we its expected to be passed from somewhere since it's a param.

Comment (jacintojosh) [reply frontend/src/states/playerStore.tsx:13]: Repeat for the setters below.

File: frontend/src/states/roomStore.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+export const RoomCodes = jennyRoomCode() as Generator;
+
+export const findPlayer = (newPlayer: Player, currentPlayers: Player[]): Player | undefined => {
+  return currentPlayers.find((currentPlayer) => currentPlayer.id === newPlayer.id);
+};
+
+const roomStore = create<RoomStore>((set: SetState<RoomStore>, get: GetState<RoomStore>) => ({
+  clientState: ClientState.Waiting,
+  players: [],
+  setRoomId: (passedrooomId) => {
```
Comment (jacintojosh) [inline frontend/src/states/roomStore.tsx:34]: See comment on playerStore file.
Can have this as `setRoomId: (roomId) => set({ roomId  }),`

File: frontend/package.json [hunk 3]
```diff
@@ -83,7 +83,7 @@
     "eslint-plugin-prettier": "^3.3.1",
     "eslint-plugin-react": "^7.22.0",
     "eslint-plugin-react-hooks": "^4.2.0",
-    "husky": "^4.3.7",
```
Comment (jacintojosh) [inline frontend/package.json:86]: Revert

File: frontend/src/states/sample_component.tsx [hunk 4]
```diff
@@ -1,17 +1,21 @@
+import { Flex, Button } from '@chakra-ui/react';
```
Comment (jacintojosh) [inline frontend/src/states/sample_component.tsx:1]: Can be removed, please ty.

File: frontend/src/components/shared/ActionBox.tsx [hunk 5]
```diff
@@ -42,9 +59,10 @@ const ActionBox = ({
               w="80%"
               mx={{ lg: 'auto', xl: 'auto' }}
               marginRight={{ sm: 4, md: 4 }}
+              onChange={(event) => setGameRoomCode(event.target.value)}
             />
           )}
-          <Button size="lg" variant="secondary" marginTop={4}>
+          <Button size="lg" variant="secondary" marginTop={4} onClick={() => ActionBoxClick(props, gameRoomCode)}>
```
Comment (jacintojosh) [inline frontend/src/components/shared/ActionBox.tsx:65]: No need to pass the whole props object if we're using only one of the properties.

File: frontend/src/components/shared/ActionBox.tsx [hunk 6]
```diff
@@ -1,16 +1,33 @@
...
+import { RoomSelections, Code } from 'types';
+
+// TODO: Join Room Function
+
+export interface ActionProps {
   actionText: string;
   headerText: string;
   hasInputField?: boolean;
-}) => {
+  roomSelection: RoomSelections;
```
Comment (jacintojosh) [inline frontend/src/components/shared/ActionBox.tsx:12]: Can be named as `roomActions`.

Comment (jacintojosh) [reply frontend/src/components/shared/ActionBox.tsx:12]: Or if we want this component to be more generic, can name as action.
Or we can actually leave this as is, and extend the generic ActionBox component into a RoomActionBox component but this would be too much work for little gains.

Honestly the above two are good in theory, but in the near future it seems we won't be creating components derived from this. So renaming this to be more readable is best.

File: frontend/src/components/shared/ActionBox.tsx [hunk 7]
```diff
@@ -1,16 +1,33 @@
...
+}
+
+const actionMap = {
+  create: createRoom,
+  join: createRoom, // TODO: should be JoinRoom
+};
+
+const ActionBoxClick = (props: ActionProps, RoomCode: Code) => {
```
Comment (jacintojosh) [inline frontend/src/components/shared/ActionBox.tsx:20]: Can be named as `HandleAction`.

File: frontend/src/components/shared/ActionBox.tsx [hunk 8]
```diff
@@ -1,16 +1,33 @@
...
+  const { roomSelection } = props;
+  const action = actionMap[roomSelection];
+  action(RoomCode);
```
Comment (jacintojosh) [inline frontend/src/components/shared/ActionBox.tsx:23]: Honestly this pattern looks iffy to me, getting a function from a map looks so weird in my pov. But can leave this for now. Jsyk.

File: frontend/src/types.ts [hunk 9]
```diff
@@ -26,17 +15,80 @@ export enum CardTypes {
   Perk = 'perk',
 }
 
+export interface Decks {
+  redDeck: Card[];
+  whiteDeck: Card[];
+}
+
+export type Id = string;
```
Comment (jacintojosh) [inline frontend/src/types.ts:23]: No need to type these separately.

**Review by jacintojosh — APPROVED** [review]
Other than the comments, LGTM!
Can merge after addressing comments.

File: frontend/src/components/shared/ActionBox.tsx (hunk 6 above)
Comment (ian-tabs) [reply frontend/src/components/shared/ActionBox.tsx:12]: to be includedd in tech debt

File: frontend/src/components/shared/ActionBox.tsx (hunk 8 above)
Comment (ian-tabs) [reply frontend/src/components/shared/ActionBox.tsx:23]: to be includedd in tech debt
